auth0/resource_auth0_client.go

```diff
@@ -330,3 +323,56 @@ func buildClient(d *schema.ResourceData) *management.Client {
 
 	return c
 }
+
+func buildClientSchema(c *management.Client, d *schema.ResourceData) {
+	d.SetId(c.ClientID)
+	d.Set("client_id", c.ClientID)
+	d.Set("client_secret", c.ClientSecret)
+	d.Set("name", c.Name)
+	d.Set("description", c.Description)
+	d.Set("app_type", c.AppType)
+	d.Set("logo_uri", c.LogoURI)
+	d.Set("is_first_party", c.IsFirstParty)
+	d.Set("oidc_compliant", c.OIDCConformant)
+	d.Set("callbacks", c.Callbacks)
+	d.Set("allowed_logout_urls", c.AllowedLogoutURLs)
+	d.Set("allowed_origins", c.AllowedOrigins)
+	d.Set("web_origins", c.WebOrigins)
+	d.Set("sso", c.SSO)
+	d.Set("sso_disabled", c.SSODisabled)
+	d.Set("cross_origin_auth", c.CrossOriginAuth)
+	d.Set("cross_origin_loc", c.CrossOriginLocation)
+	d.Set("custom_login_page_on", c.CustomLoginPageOn)
+	d.Set("custom_login_page", c.CustomLoginPage)
+	d.Set("custom_login_page_preview", c.CustomLoginPagePreview)
+	d.Set("form_template", c.FormTemplate)
+	d.Set("token_endpoint_auth_method", c.TokenEndpointAuthMethod)
+
+	if c.JWTConfiguration != nil {
+		d.Set("jwt_configuration", []map[string]interface{}{
+			{
+				"alg": c.JWTConfiguration.Algorithm,
+				"lifetime_in_seconds": c.JWTConfiguration.LifetimeInSeconds,
+				"secret_encoded":      c.JWTConfiguration.SecretEncoded,
+				"scopes":              c.JWTConfiguration.Scopes,
+			}})
+	}
+
+	if c.EncryptionKey != nil {
+		d.Set("encryption_key", c.EncryptionKey)
+	}
```

Comment:
I'm not sure these will work. Did you test/verify? Cause `encryption_key` is a `*schema.TypeSet` and we're doing a `d.Set(...)` with `c.EncryptionKey` which is of a [different type](https://github.com/yieldr/go-auth0/blob/master/management/client.go#L44). 

I'll do some testing locally and see if I can help out with this. 

The client resource is probably the most complex and full of cases we don't come across in other resources.

As a thought, in making these repetitive type casting we could learn something from the aws provider. They defined some [flattenXXX](https://github.com/terraform-providers/terraform-provider-aws/blob/ce63b929535339d463989aaa2e377a29998b0530/aws/structure.go#L764-L770) and [expandXXX](https://github.com/terraform-providers/terraform-provider-aws/blob/ce63b929535339d463989aaa2e377a29998b0530/aws/structure.go#L745-L754) functions to deal with this. Of course we can do this another time.